spec/unit/util/log_spec.rb

```diff
@@ -256,9 +289,108 @@ def log_notice(message)
       log = logs[0]
       expect(log.message).to_not match('/log_spec.rb')
       expect(log.backtrace).to be_a(Array)
+      expect(log.backtrace.first).to match('/log_spec.rb')
+    end
+
+    it "backtrace has interleaved PuppetStack when logging ParseErrorWithIssue" do
+      logs = []
+      destination = Puppet::Test::LogCollector.new(logs)
+      Puppet::Util::Log.newdestination(destination)
```

Comment:
puppet's spec_helper creates a `LogCollector` and makes it available to each rspec example as `@logs` instance variable. So I think you can drop the part above, and below do `expect(@logs.size).to eq(1)`

Reply:
Oh, nice!